src/transform/src/extractor.ts

```diff
@@ -16,23 +16,48 @@ import {
   TypeInfo,
   typeMap,
 } from "./types.js";
+import HypermodeTransform from "./index.js";
 
 export class Extractor {
   binaryen: typeof binaryen;
   module: binaryen.Module;
   program: Program;
+  transform: Transform;
```

Comment:
```suggestion
  transform: HypermodeTransform;
```
Then you don't need to check it later.